SimPEG/electromagnetics/base_1d_stitched.py

```diff
@@ -0,0 +1,544 @@
+from scipy.constants import mu_0
+import numpy as np
+from ..simulation import BaseSimulation
+from .. import props
+from .. import utils
+from ..utils.code_utils import (
+    validate_integer,
+    validate_ndarray_with_shape,
+    validate_type,
+)
+
+###############################################################################
+#                                                                             #
+#                             BaseStitchedEM1DSimulation                      #
+#                                                                             #
+###############################################################################
+
+__all__ = ["BaseStitchedEM1DSimulation"]
+
+
+class BaseStitchedEM1DSimulation(BaseSimulation):
+    """
+    Base class for the stitched 1D simulation. This simulation models the EM
+    response for a set of 1D EM soundings.
+    """
+
+    _formulation = "1D"
+    # _coefficients = []
+    # _coefficients_set = False
+
+    # _Jmatrix_sigma = None
+    # _Jmatrix_height = None
+    # _J = None
+
+    # Properties for electrical conductivity/resistivity
+    sigma, sigmaMap, sigmaDeriv = props.Invertible(
+        "Electrical conductivity at infinite frequency (S/m)"
+    )
+
+    eta = props.PhysicalProperty("Intrinsic chargeability (V/V), 0 <= eta < 1")
+    tau = props.PhysicalProperty("Time constant for Cole-Cole model (s)")
+    c = props.PhysicalProperty("Frequency Dependency for Cole-Cole model, 0 < c < 1")
+
+    # Properties for magnetic susceptibility
+    mu, muMap, muDeriv = props.Invertible(
+        "Magnetic permeability at infinite frequency (SI)"
+    )
+    chi = props.PhysicalProperty(
+        "DC magnetic susceptibility for viscous remanent magnetization contribution (SI)"
+    )
+    tau1 = props.PhysicalProperty(
+        "Lower bound for log-uniform distribution of time-relaxation constants for viscous remanent magnetization (s)"
+    )
+    tau2 = props.PhysicalProperty(
+        "Upper bound for log-uniform distribution of time-relaxation constants for viscous remanent magnetization (s)"
+    )
+
+    # Additional properties
+    h, hMap, hDeriv = props.Invertible("Receiver Height (m), h > 0")
+
+    thicknesses, thicknessesMap, thicknessesDeriv = props.Invertible(
+        "layer thicknesses (m)"
+    )
+
+    def __init__(
+        self,
+        sigma=None,
+        sigmaMap=None,
+        thicknesses=None,
+        thicknessesMap=None,
+        mu=mu_0,
+        muMap=None,
+        h=None,
+        hMap=None,
+        eta=None,
+        tau=None,
+        c=None,
+        dchi=None,
+        tau1=None,
+        tau2=None,
+        fix_Jmatrix=False,
+        topo=None,
+        parallel=False,
+        n_cpu=None,
```

Comment:
Starting to be a pretty long list of arguments. We should revisit this later and try to reduce complexity

Reply:
I am open to suggestions.